apps/coscrad-frontend/src/configurable-front-matter/validation/get-contstraint-function-for-coscrad-data-type.ts

```diff
@@ -1,34 +1,16 @@
 import { CoscradDataType } from '@coscrad/api-interfaces';
-import { isNotEmptyObject } from 'class-validator';
+import {
+    getConstraintNamesForCoscradDataType,
+    isConstraintSatisfied,
+} from '@coscrad/validation-constraints';
 
 type ConstraintFunction = (input: unknown) => boolean;
 
-type LookupTable = Record<string, ConstraintFunction>;
+export const getConstraintFunctionForCoscradDataType = (
+    dataType: CoscradDataType
+): ConstraintFunction => {
+    const constraints = getConstraintNamesForCoscradDataType(dataType);
 
-// TODO share with backend lib
-const isStringWithNonzeroLength = (input: unknown): input is string =>
-    typeof input === 'string' && input.length > 0;
-
-const lookupTable: LookupTable = {
-    [CoscradDataType.CompositeIdentifier]: isStringWithNonzeroLength,
-    [CoscradDataType.ISBN]: isStringWithNonzeroLength,
-    [CoscradDataType.NonEmptyString]: isStringWithNonzeroLength,
-    [CoscradDataType.NonEmptyString]: isStringWithNonzeroLength,
-    [CoscradDataType.NonNegativeFiniteNumber]: isStringWithNonzeroLength,
-    [CoscradDataType.PositiveInteger]: isStringWithNonzeroLength,
-    [CoscradDataType.RawData]: isNotEmptyObject,
-    [CoscradDataType.URL]: isStringWithNonzeroLength,
-    [CoscradDataType.UUID]: isStringWithNonzeroLength,
-    [CoscradDataType.Year]: isStringWithNonzeroLength,
-    [CoscradDataType.BOOLEAN]: (input: unknown) => typeof input === 'boolean',
-};
-
-export const getConstraintFunctionForCoscradDataType = (dataType: string): ConstraintFunction => {
-    const lookupResult = lookupTable[dataType];
-
-    if (lookupResult === null || typeof lookupResult === 'undefined') {
-        throw new Error(`Failed to find a validation function for COSCRAD data type: ${dataType}`);
-    }
-
-    return lookupResult;
+    return (input: unknown) =>
+        constraints.every((constraintName) => isConstraintSatisfied(constraintName, input));
```

Comment:
What if we wanted a list of constraints that are not satisfied instead?

Reply:
We can cross that bridge when we actually implement client-side form validation.